src/main/java/thePackmaster/patches/odditiespack/PackmasterFoilPatches.java

```diff
@@ -0,0 +1,192 @@
+//                oldShader = sb.getShader();
+//                sb.setShader(shade);
+//                oldColor = sb.getColor();
+//                sb.setColor(hslcBackground);
+//            }
+//        }
+//
+//        public static void Postfix(SingleCardViewPopup __instance, SpriteBatch sb) {
+//            AbstractCard card = ReflectionHacks.getPrivate(__instance, SingleCardViewPopup.class, "card");
+//            if (isFoil(card) && card.color != TheFishing.Enums.FISHING_COLOR && card.color != AbstractCard.CardColor.COLORLESS) {
+//                sb.setShader(oldShader);
+//                sb.setColor(oldColor);
+//            }
+//        }
+//    }
+//
+//    @SpirePatch(
+//            clz = SingleCardViewPopup.class,
+//            method = "renderPortrait"
+//    )
+//    public static class FoilSpecialArtSingleCardView {
+//        private static ShaderProgram oldShader;
+//        private static Color oldColor;
+//
+//        public static void Prefix(SingleCardViewPopup __instance, SpriteBatch sb) {
+//            AbstractCard card = ReflectionHacks.getPrivate(__instance, SingleCardViewPopup.class, "card");
+//            if (isFoil(card)) {
+//                oldShader = sb.getShader();
+//                sb.setShader(ART_SHADER);
+//                ART_SHADER.setUniformf("shift_amt", FoilSpecialArt.shiftAmts.computeIfAbsent(card.cardID, key -> {
+//                    Random rng = new Random((long) card.cardID.hashCode());
+//                    return 0.2F + rng.random(0F, 0.6F);
+//                }));
+//            }
+//        }
+//
+//        public static void Postfix(SingleCardViewPopup __instance, SpriteBatch sb) {
+//            AbstractCard card = ReflectionHacks.getPrivate(__instance, SingleCardViewPopup.class, "card");
+//            if (isFoil(card)) {
+//                sb.setShader(oldShader);
+//            }
+//        }
+//    }
+//
+//    @SpirePatch(
+//            clz = AbstractCard.class,
+//            method = "renderBack"
+//    )
+//    public static class FoilCardsSpecialCardbacks {
+//        private static ShaderProgram oldShader;
+//        private static Color oldColor;
+//
+//        public static void Prefix(AbstractCard __instance, SpriteBatch sb, boolean hovered, boolean selected) {
+//            if (isFoil(__instance)) {
+//                oldShader = sb.getShader();
+//                sb.setShader(shade);
+//                oldColor = ReflectionHacks.getPrivate(__instance, AbstractCard.class, "renderColor");
+//                ReflectionHacks.setPrivate(__instance, AbstractCard.class, "renderColor", hslcCardBacks);
+//            }
+//        }
+//
+//        public static void Postfix(AbstractCard __instance, SpriteBatch sb, boolean hovered, boolean selected) {
+//            if (isFoil(__instance)) {
+//                sb.setShader(oldShader);
+//                sb.setColor(oldColor);
+//            }
+//        }
+//    }
```

Comment:
Maybe remove the commented-out code?

Reply:
might uncomment it later if I have the time to work through it (tinting the card backs of foil cards)